packages/next-swc/crates/next-api/src/app.rs

```diff
@@ -894,8 +896,31 @@ impl AppEndpoint {
             ));
             server_assets.push(app_build_manifest_output);
 
+            const POLYFILL_PATH: &str = "next/dist/build/polyfills/polyfill-nomodule.js";
+            let project_path = this.app_project.project().project_path();
+            let polyfill_path = resolve(
+                project_path,
+                Value::new(ReferenceType::CommonJs(CommonJsReferenceSubType::Undefined)),
+                Request::parse_string(POLYFILL_PATH.to_string()),
+                node_cjs_resolve_options(project_path),
+            )
+            .as_raw_module_result()
+            .primary_output_assets()
```

Comment:
This is the problem.

Resolving will give you a module and not an output asset. So you want to use primary_modules here (and wrap that as RawOutput later).

Reply:
Ah, thanks! Because this method existed, I assumed these `ResolveResult` objects were more interchangeable than they are.

Regarding `RawOutput`: I don't think I can use this, because the path will point to the source file inside `node_modules/next/dist`, which cannot be resolved relative to the client's root. I think I need to use a `VirtualOutputAsset`, so that we copy the path to the `static/chunks` output directory.

On that topic, `RawOutput` isn't actually used anywhere in turbo or next... It seems like it was probably added for the sake of completeness, but now I'm wondering if it would actually ever be useful, and if not, I could make a PR to remove it.